knative-operator/pkg/controller/knativeserving/kourier/kourier.go

```diff
@@ -1,53 +1,96 @@
 package kourier
 
 import (
+	"context"
+	"fmt"
+
 	mf "github.com/jcrossley3/manifestival"
 	"github.com/openshift-knative/serverless-operator/knative-operator/pkg/common"
+	appsv1 "k8s.io/api/apps/v1"
+	v1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
 	servingv1alpha1 "knative.dev/serving-operator/pkg/apis/serving/v1alpha1"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
 var (
-	log       = common.Log.WithName("kourier")
-	manifests = []string{"deploy/resources/kourier/kourier-latest.yaml", "deploy/resources/kourier/kourier-openshift.yaml"}
+	log  = common.Log.WithName("kourier")
+	path = "deploy/resources/kourier/kourier-latest.yaml"
 )
 
 // Apply applies Kourier resources.
 func Apply(instance *servingv1alpha1.KnativeServing, api client.Client) error {
+	instance.Status.MarkDependencyInstalling("Kourier")
```

Comment:
I think this line might be the culprit for our CI issues regarding `the object has been modified; please apply your changes to the latest version and try again`. We'd set this for each time we reach this, which can even be if we have nothing to do really.

Maybe try adding the line back that I warned about (The one which checks if it's already installed) and see if that helps things.

Reply:
Note: I still think the line is wrong, but we'd need to choose a different approach here: We only want to set "MarkDependencyInstalling" if we're actually in need to reconcile things, which we have no good check for. We might need to work a bit on manifestival to enable that? While we don't have that, I'd be fine adding the check we used to have here again.

Reply:
Thank you. Yes, the line seems to have the problem. I am updating it. Perhaps we should not set status always. I found this https://github.com/kubernetes-sigs/controller-runtime/issues/741#issuecomment-571339875.